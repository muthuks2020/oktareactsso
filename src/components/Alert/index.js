import React from "react";
import PropTypes from "prop-types";
import MuiAlert from "@material-ui/lab/Alert";

const Alert = props => (
  <MuiAlert elevation={props.elevation} variant={props.variant} {...props} />
);

Alert.propTypes = {
  severity: PropTypes.string,
  elevation: PropTypes.number,
  variant: PropTypes.string,
};
Alert.defaultProps = {
  severity: "warning",
  elevation: 6,
  variant: "filled",
};

export default Alert;
