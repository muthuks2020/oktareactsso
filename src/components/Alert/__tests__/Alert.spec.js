import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Alert from "..";

describe("<Alert />", () => {
  test("should render to snapshot", () => {
    const component = renderer.create(<Alert />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should render with default props", () => {
    const wrapper = shallow(<Alert />);
    expect(wrapper.props()).toEqual({
      severity: "warning",
      elevation: 6,
      variant: "filled",
    });
  });

  test("should have Error as text", () => {
    const wrapper = shallow(<Alert severity="error">Error</Alert>);
    expect(wrapper.text()).toEqual("Error");
  });
});
