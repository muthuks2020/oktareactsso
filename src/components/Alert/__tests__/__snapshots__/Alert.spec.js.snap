// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> should render to snapshot 1`] = `
<div
  className="MuiPaper-root MuiAlert-root MuiAlert-filledWarning MuiPaper-elevation6"
  role="alert"
>
  <div
    className="MuiAlert-icon"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
      />
    </svg>
  </div>
  <div
    className="MuiAlert-message"
  />
</div>
`;
